### PR TITLE
PR (ready to be merged): OWLS-74986 Fix ItPodsRestart#testServerPodsRestartByChangingImage for external Jenkins runs

### PR DESCRIPTION
**Pull request by maggiehe00**
Now to test ITPodsRestart#testServerPodsRestartByChangeingImage we use
"container-registry.oracle.com/middleware/weblogic:12.2.1.3" --> to image:
   "container-registry.oracle.com/middleware/weblogic:duplicate" for internal env
   "container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev" for external shared cluster 
    env

Latest internal Jenkins complete test suite passed at: http://wls-jenkins.us.oracle.com/view/weblogic-operator/job/weblogic-kubernetes-operator-javatest/2001/console
External Jenkins ItPodsRestart passed at: http://build.weblogick8s.org:8080/job/weblogic-kubernetes-operator-quicktest/453/console

### CODE REVIEW

**Comment by maggiehe00** [comment]
I'm merging dev branch to this one.

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java [hunk 1]
```diff
@@ -231,26 +233,7 @@ public void testServerPodsRestartByChangingZImage() throws Exception {
 
       if (BaseTest.SHARED_CLUSTER) {
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: This check can be removed and you can simply use 
getWeblogicImageName() + ":" + getWeblogicImageDevTag()

The getWeblogicImageDevTag() method and a field can be added in BaseTest.java

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java [hunk 2]
```diff
@@ -207,7 +207,9 @@ public void testServerPodsRestartByChangingIncludeServerOutInPodLog() throws Exc
    * Modify the domain scope property on the domain resource using kubectl apply -f domain.yaml
    * Verify that all the server pods in the domain got re-started .The property tested is: image:
    * "container-registry.oracle.com/middleware/weblogic:12.2.1.3" --> image:
-   * "container-registry.oracle.com/middleware/weblogic:duplicate"
+   * "container-registry.oracle.com/middleware/weblogic:duplicate" for internal env
```
[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: Using the same image for both internal and external Jenkins was my initial thought. But yesterday as I put in k8s slack channel "container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev" failed on internal Jenkins run, which I can't figure out why.

File: integration-tests/src/test/resources/OperatorIT.properties [hunk 3]
```diff
@@ -9,5 +9,6 @@ maxIterationsPod=25
 waitTimePod=10
 weblogicImageTag = 12.2.1.3
 weblogicImageName = container-registry.oracle.com/middleware/weblogic
-OCR_SERVER="container-registry.oracle.com"
+OCR_SERVER=container-registry.oracle.com
 DOMAIN_API_VERSION = weblogic.oracle/v4
+
```
[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/resources/OperatorIT.properties]: Sure.

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (vanajamukkara) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: @maggiehe00 Locally to make the test work when container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev is used, do u have to sign/accept the agreement for this image using your login?

[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: For both local and external Jenkins run I don't need to.

[the remark this answers is not in the document]
Comment (vanajamukkara) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: what error do u see on internal jenkins? Also, can u try to re-run on external jenkins now, I have removed container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev from remote k8s cluster nodes.

[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: if you remove container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev  from remote k8s cluster nodes the test will fail, for sure.

[the remark this answers is not in the document]
Comment (vanajamukkara) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: So, the test assumes that the image already exists on the nodes? If the image is in OCR and pull image is not working, then we should fix that first. OCR user used to pull the image from internal/external jenkins may need to sign/accept the terms for  container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev.

[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: If it works for container-registry.oracle.com/middleware/weblogic:12.2.1.3 image then all other tags also should work. What is the imagePullPolicy on the CRD ?
We should try to debug in the internal Jenkins as to why it is not working.

[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: Actually not "for sure" :) I need to verify it.

[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: @vanajamukkara In fact external Jenkins run passed 
http://build.weblogick8s.org:8080/job/weblogic-kubernetes-operator-quicktest/442/console

[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: @vanajamukkara @sankarpn confirmed
imagePullPolicy: "IfNotPresent"

[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: On my local machine, I did one more time, removing all container-registry.oracle.com/middleware/weblogic images with different tags and then running the test. Also passed.

[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: can you manually pull the image before you call
TestUtils.exec(
            "docker pull container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev");

domain.verifyDomainServerPodRestart(
            "\"" + getWeblogicImageName() + ":" + getWeblogicImageTag() + "\"",
            "\"" + newImage + "\"");

[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: @sankarpn @vanajamukkara Yes, the manual pull resolved the internal Jenkins issue. 
It just occurred to me ... will we always keep "-dev" image on OCR? Original tagging image approach could be more adaptable to the images change. 
Keeping two different logic doesn't look good though.

[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: I'm going to run some test on external Jenkins.

[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: You can confirm with Monica about the -dev tag life. My guess is they planned to keep the tags.

[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsRestart.java]: Yes, Monica confirmed that we will keep -dev tag.

**Pull request merged**